packages/angular_devkit/build_angular/src/dev-server/index.ts

```diff
@@ -214,13 +231,13 @@ export class DevServerBuilder implements Builder<DevServerBuilderOptions> {
         htmlAcceptHeaders: ['text/html', 'application/xhtml+xml'],
       } as WebpackDevServer.HistoryApiFallbackConfig,
       stats: false,
-      compress: browserOptions.optimization,
+      compress: styles || scripts,
```

Comment:
Not sure, why this is not always `true`

Reply:
I think it was to make the dev server faster, by not running compression on dev builds.